### ISSUE TITLE
In-Situ 2nd-Mission Survey: Deploy to Volunteers

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#1136]
In-situ 2nd-mission survey should be served to volunteers as well as turkers. Low priority for now.

**Comment by jonfroehlich** [comment]
Given (1) that we continue to have random visitors to Project Sidewalk (despite no active advertising campaigns or community building) and (2) that this is a relatively easy add (I believe), I'd like to suggest that we boost the priority of this (though for after CSCW). It's important that we collect motivational data...

A few things:
1. The language of the survey needs to be slightly updated (see below)
2. We need to determine _when_ to show the survey and, if initially dismissed, how frequently thereafter

Re: language
![image](https://user-images.githubusercontent.com/1621749/37184276-595e130c-22ef-11e8-90c2-df218f167e01.png)

For the public-facing survey, I don't like the "task"-based language. Feels so robotic and unnatural. Instead, I suggest slight adaptations:

1. "How much have you been enjoying using Project Sidewalk?"
2. "How easy or difficult is it to use Project Sidewalk?"
3. "How well do you think you are performing on the labeling tasks?"
4. "Why did you choose to contribute to Project Sidewalk?" (no change)
5. "Do you have any feedback, design ideas, or questions?"

**Comment by manaswis** [comment]
About this point:
> We need to determine when to show the survey and, if initially dismissed, how frequently thereafter

Currently, for turkers, the survey is mandatory. It can't be dismissed unless they complete it. We would have to change this behavior for volunteers.

**Comment by jonfroehlich** [comment]
Yes

**Comment by misaugstad** [comment]
I like the modifications!

Another thing we need to think about is how we want to show the survey to anonymous users since we can't really keep track of whether or not they did the survey before.

**Comment by misaugstad** [comment]
Linking to our discussion on how to handle anonymous users in future deployments: #802

**Comment by misaugstad** [comment]
Branch 1136-make-surveys-visible-to-volunteers currently has a start to the implementation of this. It shows the survey to registered volunteers as well. However, I believe that the following is still TODO on it:

1. Show the survey to anon volunteers. Will want to wait on the implementation until we resolve #802 
2. The survey needs to be made optional for volunteers
3. The questions need to be updated, given the feedback above

**Comment by misaugstad** [comment]
This issue should be a lot easier to do now that we have changed how anonymous users are dealt with throughout the code ( PR #1293 ). Whoever addressed this issue should create a new branch for this. I see 3 things that need to be done:

1. Change the text of the survey questions
2. Allow the survey to be showed to all users (which I think will no longer be difficult)
3. Make the survey non-mandatory for non-turkers.

**Comment by jonfroehlich** [comment]
Thanks Mikey. Do you mean “addresses”?

**Comment by misaugstad** [comment]
Yes :)

**Comment by jonfroehlich** [comment]
I agree. I think everyone should have exact same survey with exact same phrasing. This is, in fact, necessary to properly compare responses across groups!

Thanks for asking. And nice catch! 👍🏽

**Comment by maubinle** [comment]
OK, I am about ready to submit the PR on this. However, I noticed some inconsistencies that it would be helpful to get clarification on, regarding the # of missions a user should complete before seeing the survey.

In the code, it says the survey should be triggered if the "numMissionsBeforeSurvey = 1." However, there is a comment accompanying this that says "NOTE the number of missions before survey is actually 3, but this check is done before the next mission is updated on the back-end" Seems to me like (even with the backend note) if we want the survey to trigger after 3 missions, we should set the numMissionsBeforeSurvey = 2. 

The second issue is, there seem to be some inconsistencies with regards to what the MissionTable.countCompletedMissionsByUserId(userId, includeOnboarding = false) function returns depending on whether or not the user has gone through the tutorial. I still need to play around with it, but sometimes the user only completes 1 mission before the survey comes up, sometimes they complete 2. This seems like it should be fixed, as it could potentially impact future research results (if they rely on survey info).

So, in summation, the questions I am asking are:
1. How many missions is the desired number of missions for the user to complete before the survey comes up, 2 or 3?
2. Should the tutorial be counted as a mission? (e.g. if the survey should come up after two missions, it could either come up after tutorial + mission 1, or after mission 1 + mission 2) Or should it be excluded from the mission count? (this seems to be, in fact, what the function above is trying to achieve with "includeonboarding = false.)
3. How important is the fact that some users see the survey after only one mission, and some see it after 2? I'm happy to fix it as part of this PR, or I can submit the PR, then fix this bug separately.

**Comment by jonfroehlich** [comment]
Unfortunately I cannot answer all your thoughtful questions. A few clarifications. 

1. All user groups should see survey at the same time
2. The survey should come up at some point after they complete their first real mission. (It  would likely be better to do this after their second real mission but our drop off rates are so high right now that I feel like we would get a biased sample—maybe that doesn’t matter?)
3. I wonder if part of the confusion is that the tutorial may be counted as a mission? I have always framed the tutorial as a guided mission and I believe you get “labeling credit” for the labels you apply in the tutorial.

**Comment by jonfroehlich** [comment]
When does it show exactly? Along with the mission completed dialog?

**Comment by maubinle** [comment]
It shows in modal mode on top of the mission completed dialog.

![image](https://user-images.githubusercontent.com/16949591/49689926-815ef300-fadd-11e8-8273-ab02deb650c0.png)

**Comment by jonfroehlich** [comment]
Immediately or after some delay?

**Comment by maubinle** [comment]
They load at the same time, so you can't really tell that the background says the mission has been completed. I could add some delay to make that more apparent?

**Comment by jonfroehlich** [comment]
My feeling is that this survey should actually come up at some point _during_ the second mission (e.g., like ~100 or 200 feet into the 2nd mission).

At the very least, I guess I don't think it should come up during the first mission complete screen as this could be disorienting the user (having not seen any previous mission complete dialogs, etc.).

**Comment by misaugstad** [comment]
We discussed and decided this is something we would like to test, so I created an issue for it #1362 but the current issue will be resolved once we have survey deployed to everyone

**Comment by misaugstad** [comment]
Closing via #1361
